qiskit_experiments/framework/base_experiment.py

```diff
@@ -265,31 +265,92 @@ def _finalize(self):
         """
         pass
 
-    def _run_jobs(self, circuits: List[QuantumCircuit], **run_options) -> List[Job]:
-        """Run circuits on backend as 1 or more jobs."""
+    def _max_circuits(self, backend: Backend = None):
+        """
+        Calculate the maximum number of circuits per job for the experiment.
+        """
+
+        if backend is None:
+            if self.backend is None:
+                raise QiskitError("A backend must be provide.")
+            backend = self.backend
+        else:
+            self._set_backend(backend)
         # Get max circuits for job splitting
-        max_circuits_option = getattr(self.experiment_options, "max_circuits", None)
+        max_circuits_option = getattr(self.experiment_options,
+                                      "max_circuits",
+                                       None)
         max_circuits_backend = self._backend_data.max_circuits
+
         if max_circuits_option and max_circuits_backend:
-            max_circuits = min(max_circuits_option, max_circuits_backend)
+            return min(max_circuits_option, max_circuits_backend)
         elif max_circuits_option:
-            max_circuits = max_circuits_option
+            return max_circuits_option
         else:
-            max_circuits = max_circuits_backend
+            return max_circuits_backend
 
-        # Run experiment jobs
-        if max_circuits and len(circuits) > max_circuits:
-            # Split jobs for backends that have a maximum job size
-            job_circuits = [
-                circuits[i : i + max_circuits] for i in range(0, len(circuits), max_circuits)
-            ]
+    def job_info(
+            self,
+            circuits: List[QuantumCircuit],
```

Comment:
This is unnecessary. The circuits should be taken from the `circuits()` method of the experiment class itself.

Reply:
I have already fixed it locally I will push it in the next commit.